src/tree/updater_colmaker.cc

```diff
@@ -202,6 +202,8 @@ class ColMaker: public TreeUpdater {
         }
         // mark subsample
         if (param_.subsample < 1.0f) {
+          CHECK_EQ(param_.sampling_method, TrainParam::kUniform)
+            << "Only uniform sampling is supported";
```

Comment:
Or `gradient based sampling is only supported by GPU Hist.`

Reply:
Done.